torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -527,6 +554,7 @@ class ScriptModuleSerializer {
   }
 
   void writeByteCode(const Module& module, bool save_mobile_debug_info) {
+    use_tensors_archive_table_ = true;
```

Comment:
Would the state of ScriptModuleSerializer be changed here? What happens if writeArchive("constants") is called after wrteByteCode()?

Reply:
I'd suggest to have use_tensors_archive_table_ as an argument to writeArchive(), instead of a member of ScriptModuleSerializer. It looks risky to me to have state of ScriptModuleSerializer changed in some hidden place like here.

Reply:
> I update the logic to use `use_tensors_archive_table_` (default to `true`), `tensors_archive_table_resources_` and `tensors_archive_table_users_` to make the logic clear. `tensors_archive_table_users_` is the archive name, which will refer to the tensors from `tensors_archive_table_resources_`. Currently the only user is `bytecode` archive and the only resource is `constants`.

As discussed offline, constants is the only resource and bytecode is the only user. We don't see solid future extension. It looks more complicated logic to me to add additional members to the class. In addition, the implicit state of use_tensors_archive_table_ is still there. I'd suggest to pass it as an arg of writeArchive(). By default it can be false, so we don't need to change all call sites.

Reply:
Address the comment in the latest commit. I think we can do this as short term solution. My concern is that `writeArchive` is supposed to be a generic function. With this pr, it includes some specific logic only for some archives (`constants` and `bytecode` in this pr). I'm afraid that it will not scale well when more logics are added to this function (someone might think that it doesn't hurt to have more specific logic in the generic function, after they see the if-statement here) and we should think of some way to make this logic more generic as follow up.